omc3/utils/outliers.py

```diff
@@ -113,8 +113,16 @@ def get_filter_mask(data: ArrayLike, x_data: ArrayLike = None, limit: float = 0.
 
 def _get_data_without_slope(mask, x, y):
     """ Remove the slope on the data by performing a linear fit. """
-    m, b = np.polyfit(x[mask], y[mask], 1)
-    return y[mask] - b - m * x[mask], y - b - m * x
+    import warnings
+    try:
+        from numpy.exceptions import RankWarning as IgnoredWarning
+    except ImportError:  # it does not exist on numpy versions supporting 3.9
+        IgnoredWarning = Warning
+    # We filter out the "Polyfit may be poorly conditioned" warning
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=IgnoredWarning)
+        m, b = np.polyfit(x[mask], y[mask], 1)
+        return y[mask] - b - m * x[mask], y - b - m * x
```

Comment:
This is similar to the `scipy` `OptimizeWarning` in another file, but ignores the `RankWarning` from `numpy` for the same reason, there isn't much we can do about it.

Caveat is that this warning only appeared in the first `numpy` version to not support Python 3.9 so some of our CI workflows would fail here. I add a hacky check.

**Note**: We might decide to keep this.

Reply:
catch and log as well